tools/RAiDER/utilFcns.py

```diff
@@ -100,51 +98,47 @@ def ecef2enu(xyz, lat, lon, height):
     return np.stack((e, n, u), axis=-1)
 
 
-def gdal_extents(fname):
+def rio_extents(fname):
     if os.path.exists(fname + '.vrt'):
         fname = fname + '.vrt'
-    try:
-        ds = gdal.Open(fname, gdal.GA_ReadOnly)
-    except Exception:
-        raise OSError('File {} could not be opened'.format(fname))
 
-    # Check whether the file is georeferenced
-    proj = ds.GetProjection()
-    gt = ds.GetGeoTransform()
+    with rasterio.open(fname) as src:
+        gt = src.transform.to_gdal()
+        proj = src.crs
+        xSize = src.width
+        ySize = src.height
+
     if not proj or not gt:
         raise AttributeError('File {} does not contain geotransform information'.format(fname))
 
-    xSize, ySize = ds.RasterXSize, ds.RasterYSize
-
     return [gt[0], gt[0] + (xSize - 1) * gt[1] + (ySize - 1) * gt[2], gt[3], gt[3] + (xSize - 1) * gt[4] + (ySize - 1) * gt[5]]
 
 
-def gdal_open(fname, returnProj=False, userNDV=None):
+def rio_open(fname, returnProj=False, userNDV=None, band=None):
     if os.path.exists(fname + '.vrt'):
         fname = fname + '.vrt'
-    try:
-        ds = gdal.Open(fname, gdal.GA_ReadOnly)
-    except BaseException:  # TODO: Which error(s)?
-        raise OSError('File {} could not be opened'.format(fname))
-    proj = ds.GetProjection()
-    gt = ds.GetGeoTransform()
 
     val = []
-    for band in range(ds.RasterCount):
-        b = ds.GetRasterBand(band + 1)  # gdal counts from 1, not 0
-        data = b.ReadAsArray()
-        if userNDV is not None:
-            logger.debug('Using user-supplied NoDataValue')
-            data[data == userNDV] = np.nan
-        else:
-            try:
-                ndv = b.GetNoDataValue()
-                data[data == ndv] = np.nan
-            except BaseException:  # TODO: Which error(s)?
-                logger.debug('NoDataValue attempt failed*******')
-        val.append(data)
-        b = None
-    ds = None
+    with rasterio.open(fname) as src:
+        proj = src.crs
+        gt = src.transform.to_gdal()
+        bnds = range(src.count)
+        if band is not None:
+            bnds = [band - 1]
+
+        for bnd in bnds:
+            data = src.read(bnd+1)
+            if userNDV is not None:
+                logger.debug('Using user-supplied NoDataValue')
+                data[data == userNDV] = np.nan
+            else:
+                try:
+                    ndv = src.nodatavals[bnd]
+                    data[data == ndv] = np.nan
+                except BaseException:  # TODO: Which error(s)?
+                    logger.debug('NoDataValue attempt failed*******')
+            val.append(data)
+    src.close()
```

Comment:
`src.close()` is not needed.

The val list is not needed either

```
with rasterio.open(fname) as ds:
   data = ds.read().transpose(1, 2, 0)
```

Reply:
Done.